core/app/models/spree/shipment.rb

```diff
@@ -214,23 +214,29 @@ def refresh_rates(shipping_method_filter = ShippingMethod::DISPLAY_ON_FRONT_END)
       return [] unless can_get_rates?
 
       # StockEstimator.new assigment below will replace the current shipping_method
-      original_shipping_method_id = shipping_method.try(:id)
+      original_shipping_method_id = shipping_method.try!(:id)
 
-      self.shipping_rates = Stock::Estimator.new(order).
+      new_rates = Stock::Estimator.new(order).
       shipping_rates(to_package, shipping_method_filter)
```

Comment:
@Mafi88 New rates are being assigned later in this change `self.shipping_rates = new_rates`. I hope this is what you are referring to.